lib/db.js

```diff
@@ -512,19 +514,30 @@ module.exports = (
       uaDeviceType: token.uaDeviceType,
       lastAccessTime: token.lastAccessTime,
       createdAt: token.createdAt
-    }]
+    }
     let sessionTokens = []
+    return getGeoData(ip)
+    .then((res) => {
+      const state = res.location && res.location.state
+      const country = res.location && res.location.country
+      newToken.location = {state, country}
+    })
+    .catch(err => {
+      newToken.location = null
+    })
     // get the array of session tokens associated with the given uid
-    return this.redis.getAsync(uid)
+    .then(() => this.redis.getAsync(uid))
```

Comment:
Actually if redis is disabled this function simply returns an empty promise : https://github.com/mozilla/fxa-auth-server/blob/master/lib/db.js#L501